File: app/src/main/java/ca/rmen/android/poetassistant/databinding/BindingCallbackAdapter.java
package ca.rmen.android.poetassistant.databinding;

import android.databinding.Observable;
import android.support.annotation.NonNull;

// http://tech.vg.no/2016/10/16/two-way-data-binding-without-rxjava/
public class BindingCallbackAdapter extends Observable.OnPropertyChangedCallback {
    private final Callback mCallback;

    public BindingCallbackAdapter(@NonNull BindingCallbackAdapter.Callback callback) {
        mCallback = callback;
    }

    @Override
    public void onPropertyChanged(Observable sender, int propertyId) {
        mCallback.onChanged();
    }

    public interface Callback {
        void onChanged();
    }
}

File: app/src/main/kotlin/ca/rmen/android/poetassistant/databinding/BindingCallbackAdapter.kt
package ca.rmen.android.poetassistant.databinding

import androidx.databinding.Observable

// http://tech.vg.no/2016/10/16/two-way-data-binding-without-rxjava/
class BindingCallbackAdapter(private val callback: Callback) : Observable.OnPropertyChangedCallback() {

    interface Callback {
        fun onChanged()
    }

    override fun onPropertyChanged(sender: Observable, propertyId: Int) {
        callback.onChanged()
    }
}
